src/org/rascalmpl/library/lang/html/IO.java

```diff
@@ -87,29 +209,66 @@ private void storeAttributes(MutableAttributeSet set) {
             for (Enumeration<?> names = set.getAttributeNames(); names.hasMoreElements(); ) {				
                 Object label = names.nextElement();
                 Object value = set.getAttribute(label);
-                attributes.peek().put(label.toString(), factory.string(value.toString()));
+                if (!"_implied_".equals(label)) {
```

Comment:
Can you add some comments on what is happening here?

Reply:
The editor kit puts _implied_ false or true on every node but that's a leaky internal implementation detail not a DOM aspect imho. We can use it to filter implied nodes but I didn't see a need for that. The implied nodes complete the dom semantically. Like a normal form.

Reply:
what is the "editor kit"?

I meant, this magic if should have an inline comment to explain the reasoning. For future you ;)